presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -306,7 +312,11 @@ public QueryInfo createQuery(Session session, String query)
             // This is intentionally not a method, since after the state change listener is registered
             // it's not safe to do any of this, and we had bugs before where people reused this code in a method
             URI self = locationFactory.createQueryLocation(queryId);
-            QueryExecution execution = new FailedQueryExecution(queryId, query, session, self, transactionManager, queryExecutor, e);
+            String loggedQuery = query;
+            if (loggedQuery.length() > maxQueryLength) {
+                loggedQuery = loggedQuery.substring(0, maxQueryLength) + " ...";
```

Comment:
I'm not sure we should add ellipses. Let's just truncate it to the maximum length. Otherwise, the logged text will exceed the length and will be corrupted (beyond truncation).